pkg/kf/commands/service-bindings/fix-orphaned-bindings.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicebindings
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/fatih/color"
+	"github.com/google/kf/v2/pkg/apis/kf/v1alpha1"
+	"github.com/google/kf/v2/pkg/kf/apps"
+	"github.com/google/kf/v2/pkg/kf/commands/config"
+	"github.com/google/kf/v2/pkg/kf/serviceinstancebindings"
+	"github.com/spf13/cobra"
+	"knative.dev/pkg/kmeta"
+)
+
+// NewDeleteOrphanedRoutesCommand creates a command to delete orphaned routes.
+func NewFixOrphanedBindingsCommand(
+	p *config.KfParams,
+	bindingsClient serviceinstancebindings.Client,
+	appsClient apps.Client,
+) *cobra.Command {
+	var (
+		dryRun = false
+	)
+
+	cmd := &cobra.Command{
+		Use:   "fix-orphaned-bindings",
+		Short: "Fix bindings without an app owner in a space.",
+		Long:  `Fix bindings with a missing owner reference to an App.`,
+		Example: `
+		# Identify broken bindings in the targeted space.
+		kf fix-orphaned-bindings
+
+		# Fix bindings in the targeted space.
+		kf fix-orphaned-bindings --dry-run=false
+		`,
+		Args:         cobra.ExactArgs(0),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			out := cmd.OutOrStdout()
+
+			if err := p.ValidateSpaceTargeted(); err != nil {
+				return err
+			}
+
+			fmt.Fprintf(out, "Finding orphaned bindings in Space: %s\n", p.Space)
+
+			bindings, err := bindingsClient.List(ctx, p.Space)
+			if err != nil {
+				return fmt.Errorf("failed to list Bindings: %v", err)
+			}
+
+			apps, err := appsClient.List(ctx, p.Space)
+			if err != nil {
+				return fmt.Errorf("failed to list Apps: %v", err)
+			}
+			appLookupTable := make(map[string]v1alpha1.App)
+			for _, a := range apps {
+				appLookupTable[a.Name] = a
+			}
+
+			cmdColor := color.New(color.FgCyan)
+
+			deletedBindingsCount := 0
+			ownersFixedCount := 0
+			for _, b := range bindings {
+				appRef := b.Spec.App
+				if appRef == nil {
+					// Case: binding refers to something other than an app.
+					continue
+				}
+
+				app, ok := appLookupTable[appRef.Name]
+				if !ok {
+					// Case: App has been deleted, remove binding.
+					fmt.Fprintf(out, "Deleting binding %q which refers to a missing app %q\n", b.Name, appRef.Name)
+					deletedBindingsCount++
```

Comment:
nit: should this be named to something like bindingsToDelete, and increase deletedBindingsCount only when the actual deletion succeeded? So that user won't get confused when doing dry run